spotify_db/Query.sql

```diff
@@ -0,0 +1,169 @@
+-- Top popular tracks
+SELECT t.track_name, ar.artist_name, al.album_name, t.popularity
+FROM tracks t
+JOIN artists ar ON t.artist_id = ar.artist_id
+JOIN albums al ON t.album_id = al.album_id
+WHERE t.popularity > (SELECT AVG(popularity) FROM tracks)
+ORDER BY t.popularity DESC; 
+
+-- Artists 
+SELECT 
+    ar.artist_name,
+    ar.followers,
+    COUNT(DISTINCT t.track_id) AS track_count,
+    COUNT(DISTINCT t.album_id) AS album_count,
+    COUNT(DISTINCT pt.playlist_id) AS playlist_count,
+    ar.popularity AS artist_popularity,
+    AVG(t.popularity) AS avg_track_popularity,
+    AVG(al.popularity) AS avg_album_popularity
+FROM artists ar
+LEFT JOIN tracks t ON ar.artist_id = t.artist_id
+LEFT JOIN albums al ON t.album_id = al.album_id
+LEFT JOIN playlist_tracks pt ON t.track_id = pt.track_id
+GROUP BY ar.artist_id
+ORDER BY ar.followers DESC;
+
+-- Most Popular Genres by Artists
+SELECT genre, COUNT(*) AS genre_count
+FROM artist_genres
+GROUP BY genre
+ORDER BY genre_count DESC;
+
+-- Genres Popularity Based on Track, Artist and Album Popularity
+SELECT 
+    arg.genre,
+    COUNT(DISTINCT t.artist_id) AS total_artists,  
+    AVG(t.popularity) OVER (PARTITION BY arg.genre, t.artist_id) AS avg_artist_popularity,  
+    COUNT(*) AS total_tracks,  
+    AVG(t.popularity) AS avg_track_popularity,  
+    COUNT(DISTINCT t.album_id) AS total_albums,  
+    AVG(t.popularity) OVER (PARTITION BY arg.genre, t.album_id) AS avg_album_popularity  
+FROM tracks t
+JOIN artist_genres arg ON CAST(t.artist_id AS INTEGER) = arg.artist_id
+GROUP BY arg.genre
+ORDER BY total_tracks DESC;
+
+-- Tracks by Release Year
+SELECT SUBSTR(al.release_date, 1, 4) AS release_year, COUNT(t.track_id) AS num_tracks
+FROM tracks t
+JOIN albums al ON t.album_id = al.album_id
+GROUP BY release_year
+ORDER BY release_year;
+
+-- Albums by Release Year
+SELECT SUBSTR(al.release_date, 1, 4) AS release_year, COUNT(DISTINCT al.album_id) AS num_albums
+FROM albums al
+GROUP BY release_year
+ORDER BY release_year;
```

Comment:
These queries LGTM. Though I'm not clear how they fit into your analysis's goal.